src/components/data/WdsTroubleshooter.js

```diff
@@ -0,0 +1,153 @@
+    ['success', () => icon('success-standard', { size, style: { color: colors.success() }, 'aria-label': 'Validation Success' })],
+    ['failure', () => icon('error-standard', { size, style: { color: colors.danger(0.85) }, 'aria-label': 'Validation Failure' })],
+    ['running', () => icon('loadingSpinner', { size, style: { color: colors.primary() }, 'aria-label': 'Validation Running' })]
+  )
+
+  const clippy = (label, value) => !!value && h(Fragment, [value, h(ClipboardButton, {
+    'aria-label': `Copy ${label} to clipboard`,
+    style: { marginLeft: '1rem' },
+    text: value
+  })])
+
+  const troubleShooterRow = ([label, text, iconRunning, iconSuccess, copy, element]) => {
+    return tr([
+      td({ style: { fontWeight: 'bold' } }, [
+        iconRunning ? checkIcon('running') : (iconSuccess ? checkIcon('success') : checkIcon('failure'))
+      ]),
+      td({ style: { fontWeight: 'bold' } }, [label]),
+      td({ style: { fontWeight: 'bold' } }, [!!element]),
+      Utils.cond([!!element, () => element], () => Utils.cond([copy, () => td([clippy(label, text)])], () => td([h(Fragment, [text])]))
+      )
+    ])
+  }
+
+  // The proxyUrl is long and should be truncated, but
+  // we still want to be able to copy it, so it gets it own special element
+  // that doesn't use the clippy() helper
+  const proxyElement =
+      td([
+        h(Fragment, [h(div, { style: _.merge({ width: '400px', float: 'left' }, Style.noWrapEllipsis) }, [proxyUrl]),
+          h(ClipboardButton, {
+            'aria-label': 'Copy proxy url to clipboard',
+            style: { marginLeft: '1rem' },
+            text: proxyUrl
+          })])
+      ])
+
+  /** For each piece of information we want to include in the troubleshooter, we want:
+   * 1. A label
+   * 2. The value of information
+   * 3. A function or variable that evaluates to boolean, determining whether the info/validation is still running
+   * 4. A function or variable that evaluates to boolean, determining whether the info/validation is successful
+   * 5. A boolean that determines whether or not to display a copy-clipboard icon for this piece of info
+   * 6. An optional element to use in place of the standard defined in troubleShooterRow
+  **/
+  const troubleShooterText = [
+    ['Workspace Id',	workspaceId, false, !!workspaceId, true],
+    ['Resource Group Id',	mrgId, false, !!mrgId, true],
+    ['App listing',	`${leoOk?.length} app(s) total`, leoOk == null, !!leoOk?.length, false],
+    ['Data app name',	appFound, leoOk == null, !!appFound && appFound !== 'unknown', true],
+    ['Data app running?',	appRunning, appRunning == null, !!appRunning && appRunning !== 'unknown', false],
+    ['Data app proxy url',	proxyUrl, proxyUrl == null, !!proxyUrl && proxyUrl !== 'unknown', false, proxyElement],
+    ['Data app responding',	JSON.stringify(wdsResponsive), wdsResponsive == null, !!wdsResponsive && wdsResponsive !== 'unknown', false],
```

Comment:
Not sure what to do about values here that are will evaluate to be true or false (here, `wdsResponsive` and `defaultInstanceExists`.  If I stringify them then while they are in the "unknown" state they show up with extra quotes around them:
<img width="883" alt="image" src="https://user-images.githubusercontent.com/5884792/215114857-311744b3-e73e-40eb-b8ff-98d0cfe0a904.png">

But if I don't stringify them then when they are true or false they just don't show up at all:
<img width="890" alt="image" src="https://user-images.githubusercontent.com/5884792/215115013-09bc6023-60a3-41b2-9ddf-7c567a8b55cf.png">
Before I spent hours fine-tuning this, is there a simple fix?

Reply:
Yes!  Thank you!! 🦸